lib/MenuItem.php

```diff
@@ -107,8 +104,16 @@ public static function build( $data, Menu $menu = null ) : self {
 	 * @param \Timber\Menu $menu The `Timber\Menu` object the menu item is associated with.
 	 */
 	protected function __construct( $data, $menu = null ) {
+		$this->core_object = $data;
 		$this->menu = $menu;
+
 		$data       = (object) $data;
+
+		// Items from WordPress can come with a $title property which conflicts with methods
+		$data->__title = $data->title;
+		// @todo Use case for unsetting title?
+		// unset($data->title);
```

Comment:
@gchtr @jarednova Is it really needed? This prevents nav walker to print menu titles.

Reply:
Found the issue where the `$__title` property was introduced: https://github.com/timber/timber/issues/334.

The problem was when an ACF field named `title` existed. These kind of conflicts lead to the introduction of using `meta()` as the official way to get meta data from an object instead of directly using the field’s name. So if you have an ACF field for a menu item called title and you want to access it, you should use `{{ item.meta('title') }}`  instead of `{{ item.title }}`, because Timber wouldn’t know what to use.

We can definitely get rid of the `$__title` property and use `$title` directly.

Thanks for flagging this, @nlemoine!

Reply:
Removing it is not so easy, because we need Twig to call the `title()` method in order to apply the `nav_menu_item_title`. If `$item->title` becomes a public property, `{{ item.title }}` won't call `title()` and no filter will be applied.

I added a `$title` and marked it as protected to solve this.

Reply:
@nlemoine Thanks for the update! Would it make sense to completely get rid of the `$title` property? Could this be a use case for #2560?

If we use `{{ item.title }}`, then the `title()` method would be called, were we could use `$this->wp_object->title` instead of `$this->title`:

```php
$title = apply_filters( 'nav_menu_item_title', $this->wp_object->title, $this->wp_object, $this->menu->args ? $this->menu->args : new \stdClass, $this->level );
```

Or might I be missing something?

Reply:
Good point! I see you’ve thought this through. Both solutions are fine for me then, too.

Reply:
So change is required I guess :) Let me know if you still see something, the PR is ready to merge from my side.